### PR TITLE
Api Endpoint

### PR DESCRIPTION
**Pull request by sheezu900**
This is a collaborative project and participants are:
Shah Zaman & @Graciano1997 

### Within this branch we have:

- Created an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,6 +1,16 @@
 class CommentsController < ApplicationController
   load_and_authorize_resource
 
+  def index
+    @post = Post.find(params[:post_id].to_i)
+    @posts_comments = @post.comments
```
Comment (ademibowale) [inline app/controllers/comments_controller.rb]: - [ ] You did a great job so far ,i will recomend you to  add /api to URL or api/v1 with version number in order to identify as an API based URL, 
- [ ] kindly generate a controler for api in a professional way there should be a Api/v1 folder in your controller folder This is a common convention used to distinguish between different versions of an API. As an API evolves over time, changes may be made to the endpoints, response formats, or request parameters.  see example below 

Screenshot 2023-11-18 221640

**Review by ademibowale — CHANGES_REQUESTED** [review]
# Required Changes 🔁
# Hello @sheezu900     😄 

You're doing a great job on your project. We're proud of you. Good job so far! 

There are some issues that you still need to work on to go to the next project but you are almost there!

just a little bit

## Highlights
  ✅ Nice user interface
  ✅Correct use of Github Flow
  ✅ No linter errors
✅ Descriptive Pull Request
  

## Required Changes ♻️




_Check the comments under the review._




## Optional suggestions 
   
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (rloterh) [reply app/controllers/comments_controller.rb]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Shah & Graciano,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on introducing improvements to the api structure!

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**